demo/agilebank/templates/k8srequiredprobes_template.yaml

```diff
@@ -0,0 +1,64 @@
+apiVersion: templates.gatekeeper.sh/v1beta1
+kind: ConstraintTemplate
+metadata:
+  name: k8srequiredprobes
+spec:
+  crd:
+    spec:
+      names:
+        kind: K8sRequiredProbes
+      validation:
+        openAPIV3Schema:
+          properties:
+            probes:
+              type: array
+              items:
+                type: string
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package k8srequiredprobes
+
+        violation[{"msg": msg}] {
```

Comment:
Enforcing on Deployments/ReplicaSets, etc. runs counter to the conclusion in #298 I think there is still valid debate on whether that is the right behavior, but in the meantime we should have uniform behavior for templates.